### ISSUE TITLE
ERROR 500 ( Try to Edit findings)

### ISSUE THREAD

**Issue by topahadzi** [DefectDojo/django-DefectDojo#6453]
**Be informative**
ERROR 500 ( Try to Edit Findings)

**Bug description**
Got error when try to edit finding on engagement

**Steps to reproduce**
1. Go to 'product'
2. Click on 'engagement'
3. Click on 'Findings'
4. Click edit
5. Save With Change and got error
6. Save Without any Change and got error

**Expected behavior**

**Deployment method** *(select with an `X`)*
- [X] Docker Compose
- [ ] Kubernetes
- [ ] GoDojo

**Environment information**
 - Operating System: [e.g. Ubuntu 18.04]
 - Defectdojo Version v. 2.11.0( release mode )

**Logs** 
```
[22/Jun/2022 18:37:43] ERROR [django.request:224] Internal Server Error: /finding/122637/edit
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/app/./dojo/authorization/authorization_decorators.py", line 30, in _wrapped
    return func(request, *args, **kwargs)
  File "/app/./dojo/finding/views.py", line 816, in edit_finding
    finding_helper.save_vulnerability_ids(new_finding, form.cleaned_data['vulnerability_ids'].split())
  File "/app/./dojo/finding/helper.py", line 567, in save_vulnerability_ids
    obj, created = Vulnerability_Id.objects.get_or_create(
  File "/usr/local/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 581, in get_or_create
    return self.get(**kwargs), False
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 439, in get
    raise self.model.MultipleObjectsReturned(
dojo.models.Vulnerability_Id.MultipleObjectsReturned: get() returned more than one Vulnerability_Id -- it returned 2!
```

**Comment by StefanFl** [comment]
Thanks @topahadzi for reporting this bug.

This shouldn't be possible, but obviously is possible. #6535 changes the algorithm to save vulnerability ids and makes it more robust.

**Comment by StefanFl** [comment]
merged for release 2.13.0
